### DISCUSSION TITLE
move-column-to-workspace without automatic focus-workspace

### DISCUSSION THREAD

**Discussion by tv42** [YaLTeR/niri#457] [Ideas]
Right now, if you move a column/window to a different workspace, your focus will also be switched there.

I would like a move that doesn't change focused workspace. In fact, I would have assumed that that's what "move" means -- what is there now is "move-column-and-self".

I often send a group of related windows to a new workspace (to start a new "topic"), and if my focus changes after every window, that requires much more user input and carefulness, going back after every move.

You can hack around this with `niri msg action move-column-to-workspace 9 && niri msg action focus-workspace-previous` but that's not an easy way set up bindings and it's wasteful with subprocesses and doing extra work.

Related but not really the same thing: with the automatic focus switch, sometimes I have trouble telling whether moving a full-width column to a new workspace did anything -- nothing on the screen changes!

**Comment by YaLTeR** [comment]
It works this way because most of the time you're only moving one column, so switching focus there is also what you want. For a group of windows I guess what you could also do is consume them all into one column, then move that at once.

> sometimes I have trouble telling whether moving a full-width column to a new workspace did anything

Yeah, this problem comes up in a few places. Haven't decided yet what's the best way to deal with it.

**Comment by thairanaru** [comment]
Yeah, I also have it hacked like that. 
Another use case I have is that I somtimes want to kick out a column just out of my workspace so I can return to it later.

**Comment by itshog** [comment]
Personally I think that this option would be extremely useful, as I often open mutiple files and application on the workspace I'm in and I want to arrange some of these apps in other workspaces, without necessarily focusing these other workspace every time I move an app

**Comment by kj** [comment]
I also tend to move windows this way. It's actually very rare that I'll want to focus the window I just moved as I almost always just want to move some irrelevant windows out of the way to tidy up the current workspace and help me focus. So jumping repeatedly to the workspace I'm moving them to is very annoying.

**Comment by alexdev16** [comment]
People just want to reincarnate the action "minimize". What can you do about it..? Habit. )

**Comment by kj** [comment]
I should be more clear that I'm not just moving them out of the way (not the best choice of words), I am still arranging them in new workspaces deliberately. I don't think it's very much like minimising.

**Comment by aberter0x3f** [comment]
#1465

**Comment by tv42** [comment]
Works great in v25.05!

```
    Mod+1 { focus-workspace 1; }
    Mod+2 { focus-workspace 2; }
    Mod+3 { focus-workspace 3; }
    Mod+4 { focus-workspace 4; }
    Mod+5 { focus-workspace 5; }
    Mod+6 { focus-workspace 6; }
    Mod+7 { focus-workspace 7; }
    Mod+8 { focus-workspace 8; }
    Mod+9 { focus-workspace 9; }
    Mod+Shift+1 { move-column-to-workspace focus=false 1; }
    Mod+Shift+2 { move-column-to-workspace focus=false 2; }
    Mod+Shift+3 { move-column-to-workspace focus=false 3; }
    Mod+Shift+4 { move-column-to-workspace focus=false 4; }
    Mod+Shift+5 { move-column-to-workspace focus=false 5; }
    Mod+Shift+6 { move-column-to-workspace focus=false 6; }
    Mod+Shift+7 { move-column-to-workspace focus=false 7; }
    Mod+Shift+8 { move-column-to-workspace focus=false 8; }
    Mod+Shift+9 { move-column-to-workspace focus=false 9; }
```

**Comment by rickswe** [comment]
It looks like there's a lot of actions that do not have the focus option (see comments)?  For now using the
uncommented bindings.

```kdl
    // Mod+Ctrl+Semicolon  { move-window-to-monitor-next focus=false; }
    // Mod+Ctrl+Semicolon  { move-workspace-to-monitor-previous focus=false; }
    // Mod+Ctrl+Semicolon  { move-column-to-monitor-next focus=false; }
    // Mod+Ctrl+Slash      { move-column-to-monitor-next focus=false; }
    Mod+Ctrl+Semicolon   { spawn "sh" "-c" r#"
                                  niri msg action move-column-to-monitor-previous
                                  niri msg action focus-monitor-next"#; }
    Mod+Ctrl+Slash       { spawn "sh" "-c" r#"
                                  niri msg action move-column-to-monitor-next
                                  niri msg action focus-monitor-previous"#; }
```
